polygeist/tools/cgeist/driver.cc

```diff
@@ -294,16 +295,23 @@ static void enableOptionsPM(mlir::PassManager &pm) {
   pm.enableTiming(timing);
 }
 
+static bool isFuncOpName(llvm::StringRef FuncOpName) {
+  return FuncOpName == mlir::func::FuncOp::getOperationName();
+}
+
```

Comment:
I'd prefer that `FuncOpName` was a OperationName rather than a StringRef:
```
 static bool isFuncOp(const OperationName &FuncOpName) {
  return FuncOpName.getStringRef() == mlir::func::FuncOp::getOperationName();
}